### ISSUE TITLE
Warns about file not being saved, even after export?

### ISSUE THREAD

**Issue by steveharman** [wulkano/kap#399]
Maybe I've misunderstood this alert / dialog, but....

**macOS 10.12.6:**
**Kap Version: 2.0.0(681)**

### Steps to reproduce
Create a recording
Export > MP4 - (which seems to be working these days)
Close Kap using the red cross icon in its menubar 

### Current behaviour
Displays an alert "Are you sure you wish to discard this recording?  It will not be saved"

### Expected behaviour
As I've just 'saved' (exported) my recording this seems a little confusing?

### Workaround
n/a /  ignore it

![kap_editor](https://user-images.githubusercontent.com/12557929/36668433-36ee1f96-1ae9-11e8-82a2-3d251ed11541.png)

**Comment by skllcrn** [comment]
It refers to the original recording, if you close the window you'll lose it and no longer be able to export it. I think the language is what's confusing here, especially if you've already exported once. I suggest we change the copy to something along the lines of "You'll no longer be able to edit and export the original recording".

**Comment by steveharman** [comment]
What is the concept of 'original recording' in Kap please?  

I get the basic concept, say Sketch might have a native .sketch "original" file format used for opening & saving work and also some _export_ options such as PNG, JPG etc.   Does Kap have the concept of Saving the 'original recording' in a native KAP file format?  I couldn't find any Save options, only export.

In which case if exporting is the _only_ way of Saving my recording then I don't understand what the warning is about.  I will lose my work unless I......take what steps?

**Comment by albinekb** [comment]
The video you see in the editor is the "original recording", it is the temporary file that the actual recording library (https://github.com/wulkano/aperture) gives to Kap when the recording is ended (via menubar click). Once you close the editor this file is no longer accessible and thus the "original recording" will be  "lost". Does that make sense? 😊

**Comment by steveharman** [comment]
Yup.  Got it, thanks. 

Case; I have TextEdit open, type some words and click the close control.  I correctly get a warning about losing my work unless I save it first. Makes sense - the action I need to take to not see a warning is save my work.

What are the steps I need to take to satisfy the warning in Kap?  

I've already exported my file - what else can I  do?  There doesn't seem to be a Save option to save the original file so..... ?

**Comment by stevenpeniche** [comment]
> I suggest we change the copy to something along the lines of "You'll no longer be able to edit and export the original recording".

Could I go ahead and implement this change?

**Comment by steveharman** [comment]
@stevenpeniche I still think a point is being missed here (or I still don't understand).   It's not so much about the wording and more about what action can I take to **not** get this warning. 

Back to my example of TextEdit:  user gets a warning that they'll lose their unsaved work _unless_ they File > Save.  So they use File > Save and the warning no longer applies, they can close their document and all is well.

What steps in Kap would satisfy the "You'll lose your work" alert so user can close the Kap window _without_ seeing a warning.  

Here is my workflow within Kap:

1. Create a recording
2. Export as MP4
3. Close Kap window.

I don't need or want a warning about losing the work that I've just saved (exported).  Especially as I can't find any other way of "saving" my work.

What have I missed?  ;-)

**Comment by stevenpeniche** [comment]
@steveharman The way I see it is this.

Workflow:
1. Create a recording.
2. Export as MP4.
3. Confirm that you've exported the recording in the format you desire.
4. Close Kap window.

That third step helps validate that the user doesn't have any second thoughts in regards to size, fps or file format.

**Comment by albinekb** [comment]
Another idea I have had is to show a list of previous Kaptures, as the original file is still accessable (it's in os.tmpDir()) so we could keep a list in memory of the last kapture, that way we could add a button in the interface to export the last recorded kapture (in the same app session), maybe that's another way to help users? @skllcrn thoughts?

**Comment by willedahlbo** [comment]
Maybe ´Save...´ instead of ´Export´could help clarify this? <img width="647" alt="screen shot 2018-03-01 at 12 09 12" src="https://user-images.githubusercontent.com/2728994/36842344-b48d5dfe-1d4b-11e8-94e8-b28220124fad.png">

**Comment by LinusU** [comment]
I have also found this behavior very confusing, and I actually haven't know what I've done wrong until I read this issue.

> 3) Confirm that you've exported the recording in the format you desire.

I personally don't feel that this step is necessary. The way I would personally prefer would be to switch "export" to "save" (per @willedahlbo suggestion) and then only warn if I close the window without having used any of those buttons (e.g. I haven't exported a single file)

**Comment by steveharman** [comment]
That might help @willedahlbo In conjunction with not alerting user they may lose work *after* they've performed an export / save action.

To my mind it's:

1. Create a recording.
2. Export (or Save...) in chosen format.
3. Close Kap window without alert

And tbh I'd personally be happier that Kap did point 3 for me, automatically.  Recording > Export > End of.  Without needing to close the window.

As it stands I can't understand what risk the alert is trying to protect user from _after_ they have exported (Saved)?  User has saved their work.  End of story?  What is it they're at risk of losing?

**Comment by levrik** [comment]
@skllcrn It would be awesome to show some checkbox in this dialog which you can check if you're no longer interested in this warning. This warning is really annoying.
Actually I would also like an option for the window to automatically close after export. I can open another issue for that.

**Comment by steveharman** [comment]
Agreed with @skllcrn 

It just seems a little strange - iirc we can Cmd-Q the app and don't get warned about losing anything, yet after using the export options (to save our work to MP4, for example) we get this warning about potentially losing work. 

If the warning has validity for some users then I'm cool with that - but personally I find it annoying and a "Don't warm me again" checkbox would seem to cover all cases.

**Comment by albinekb** [comment]
I agree a checkbox would be nice here! I'm using the `discard` button on the touchbar which does not require confirmation, so I'm unaffected by this.

**Comment by skllcrn** [comment]
This behaviour is completely removed in 3.0 as we've introduced export history, allowing you to reopen the original recordings for the duration of your session.

**Comment by steveharman** [comment]
I still get an "_Are you sure that you want to discard this recording_?" alert in 3.0.0 Beta 5 @skllcrn - is that expected behaviour?   

I took it from https://github.com/wulkano/kap/issues/399#issuecomment-423510473 that the alert wouldn't be shown after the user has exported their recording?

Thanks

![kap2](https://user-images.githubusercontent.com/12557929/54611214-71e13580-4a4e-11e9-9dec-f846009353fe.png)

**Comment by karaggeorge** [comment]
I think this check is, as the message says, for the original recording. Even if you exported let's say as a gif, closing the app means that you will lose that original mp4 unedited recording (since we save it in the tmp directory). If you have an editor that is opened from a local file (i.e. by dropping a file in the icon) then you don't get that message when trying to close, since you can just re-open the file.
